example/join-from-mediadevice/main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	log "github.com/pion/ion-log"
+	sdk "github.com/pion/ion-sdk-go"
+	"github.com/pion/webrtc/v3"
+	"github.com/pion/mediadevices"
+	"github.com/pion/mediadevices/pkg/codec/vpx"
+	"github.com/pion/mediadevices/pkg/frame"
+	"github.com/pion/mediadevices/pkg/prop"
+
+	// Note: If you don't have a camera or microphone or your adapters are not supported,
+	//       you can always swap your adapters with our dummy adapters below.
+	// _ "github.com/pion/mediadevices/pkg/driver/videotest"
+	// _ "github.com/pion/mediadevices/pkg/driver/audiotest"
+	_ "github.com/pion/mediadevices/pkg/driver/camera"     // This is required to register camera adapter
+	_ "github.com/pion/mediadevices/pkg/driver/microphone" // This is required to register microphone adapter
+
+)
+
+func main() {
+	// init log
+	fixByFile := []string{"asm_amd64.s", "proc.go", "icegatherer.go"}
+	fixByFunc := []string{"AddProducer", "NewClient"}
+	log.Init("debug", fixByFile, fixByFunc)
+
+	// parse flag
+	var session, addr string
+	flag.StringVar(&addr, "addr", "localhost:50051", "Ion-sfu grpc addr")
+	flag.StringVar(&session, "session", "test room", "join session name")
+	flag.Parse()
+
+	// add stun servers
+	webrtcCfg := webrtc.Configuration{
+		ICEServers: []webrtc.ICEServer{
+			webrtc.ICEServer{
+				URLs: []string{"stun:stun.stunprotocol.org:3478", "stun:stun.l.google.com:19302"},
+			},
+		},
+	}
+
+	config := sdk.Config{
+		Log: log.Config{
+			Level: "debug",
+		},
+		WebRTC: sdk.WebRTCTransportConfig{
+			Configuration: webrtcCfg,
+		},
+	}
+	// new sdk engine
+	e := sdk.NewEngine(config)
+
+	// get a client from engine
+	c := e.AddClient(addr, session, "client id")
+
+	// client join a session
+	err := c.Join(session)
+
+	if err != nil {
+		log.Errorf("err=%v", err)
+		panic(err)
+	}
+
+
+	vpxParams, err := vpx.NewVP8Params()
+	if err != nil {
+		panic(err)
+	}
+	vpxParams.BitRate = 500_000 // 500kbps
+
+	codecSelector := mediadevices.NewCodecSelector(
+		mediadevices.WithVideoEncoders(&vpxParams),
+	)
+
+	fmt.Println(mediadevices.EnumerateDevices())
+
+	s, err := mediadevices.GetUserMedia(mediadevices.MediaStreamConstraints{
+		Video: func(c *mediadevices.MediaTrackConstraints) {
+			c.FrameFormat = prop.FrameFormat(frame.FormatYUY2)
+			c.Width = prop.Int(640)
+			c.Height = prop.Int(480)
+		},
+		Codec: codecSelector,
+	})
+
+	if err != nil {
+		panic(err)
+	}
+
+	for _, track := range s.GetTracks() {
+		track.OnEnded(func(err error) {
+			fmt.Printf("Track (ID: %s) ended with error: %v\n",
+				track.ID(), err)
+		})
+		_, err = c.Publish(track)
+		if err != nil {
+			panic(err)
+		} else {
+			break // only publish first track, thanks
```

Comment:
I'm not sure why mediadevices would return 3 tracks, but I was seeing three of myself